processManager.js

```diff
@@ -0,0 +1,57 @@
+const { spawn } = require("child_process");
+const { getLangugeByFilePath } = require('./utilities');
+const { getBinLocation } = require('./locationHelper');
+const config = require('./config');
+
+/**
+ * Execute an executable file
+ * @param {string} filePath bin or .py
+ * @param {object} options
+ */
+function execFile(filePath, options = {}) {
+    const language = getLangugeByFilePath(filePath);
+    const opts = {...options, timeout: config.timeout};
+
+    if (language === 'Python')
+        return spawn(config.compilers[language], [filePath], opts)
+    
+    const binPath = getBinLocation(filePath);
+    return spawn(binPath, opts);
+}
+
+// kills all spawned process
+function killAll(stack) {
+    console.log("Killing all spawns");
+    stack.forEach(proc => proc.kill());
+}
+
+/**
+ * Create timeout to kill the process
+ * @param {*} process 
+ * @param {object} info
+ */
+function setKiller(process, info = {}) {
+    return setTimeout(() => {
+        console.log(`${config.timeout} sec killed process`, info);
+        process.kill();
+    }, config.timeout);
+}
+
+/**
+ * Get rid of created bin if exists
+ * @param {string} filePath bin or .py
+ */
+function removeBin(filePath) {
+    const binPath = getBinLocation(filePath);
+    if (binPath) {
+        spawn("rm", [binPath]);
```

Comment:
The code here is wrong, as `getBinLocation` for python files returns nothing. Even if it did return the filepath it would be wrong, as wee need `.pyc` path. I've pushed a commit on top to fix this and some other small things.

Reply:
It wasn't wrong! The `.pyc`s are created when the file  is imported and running a script create no compiled file. So we need `getBinLocation` returns nothing (`undefined`) for Python.

Reply:
But `getBinLocation` is not used for Python at all, because you placed checks everywhere to simply return without doing anything; so the only place it used in case of python is for spawning `rm` and `del`.

Reply:
Yeah, I know. That was kind double-check for safety; Also I wanted `removeBin` works correctly anyway as a function which is responsible for removing generated binary files.

Reply:
Cool!